brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/spec/VanillaSoftwareProcessLifecycleModifier.java

```diff
@@ -65,96 +74,94 @@ public void apply(EntitySpec<?> entitySpec, NodeTemplate nodeTemplate, Topology
+                continue;
             }
+
+            Operation op = operations.get(opKey);
+            applyLifecycle(op, entitySpec, lifeCycleMapping.get(opKey), nodeTemplate, topology);
         }
     }
 
     protected Map<String, Operation> getStandardInterfaceOperations(NodeTemplate nodeTemplate, Topology topology) {
         Map<String, Operation> operations = MutableMap.of();
-        Interface indexedNodeTemplateInterface = findInterfaceOfNodeTemplate(
-                getIndexedNodeTemplate(nodeTemplate, topology).get().getInterfaces(), VALID_INTERFACE_NAMES);
-        Interface nodeTemplateInterface = findInterfaceOfNodeTemplate(
+        IndexedArtifactToscaElement indexedNodeTemplate = getIndexedNodeTemplate(nodeTemplate, topology).get();
+
+        Interface indexedNodeTemplateInterface = NodeTemplates.findInterfaceOfNodeTemplate(
+                indexedNodeTemplate.getInterfaces(), VALID_INTERFACE_NAMES);
+
+        Interface nodeTemplateInterface = NodeTemplates.findInterfaceOfNodeTemplate(
                 nodeTemplate.getInterfaces(), VALID_INTERFACE_NAMES);
 
-        if ((indexedNodeTemplateInterface != null)
-                && (indexedNodeTemplateInterface.getOperations() != null)) {
+        if (indexedNodeTemplateInterface != null && indexedNodeTemplateInterface.getOperations() != null) {
             operations = MutableMap.copyOf(indexedNodeTemplateInterface.getOperations());
         }
 
-        if ((nodeTemplateInterface != null) && (nodeTemplateInterface.getOperations() != null)) {
-            for (String operationName : nodeTemplateInterface.getOperations().keySet()) {
-                if (operations.containsKey(operationName)) {
-                    operations.put(operationName,
-                            nodeTemplateInterface.getOperations().get(operationName));
-                }
+        if (nodeTemplateInterface != null && nodeTemplateInterface.getOperations() != null) {
+            copyValues(operations, nodeTemplateInterface.getOperations());
+        }
+        return operations;
+    }
+
+    private Map<String, Operation> copyValues(Map<String, Operation> operations, Map<String, Operation> interfaceOperations) {
+        for (String operationName : interfaceOperations.keySet()) {
+            if (operations.containsKey(operationName)) {
+                operations.put(operationName, interfaceOperations.get(operationName));
             }
         }
         return operations;
     }
 
-    protected void applyLifecycle(Map<String, Operation> ops, String opKey, EntitySpec<? extends Entity> spec,
-            ConfigKey<String> cmdKey, NodeTemplate nodeTemplate, Topology topology) {
-        Operation op = ops.remove(opKey);
-        if (op == null) {
+    protected void applyLifecycle(Operation op, EntitySpec<? extends Entity> spec,
+                                  ConfigKey<String> cmdKey, NodeTemplate nodeTemplate, Topology topology) {
```

Comment:
Prefer second line to be indented twice from beginning of method signature rather than aligned with first argument. IntelliJ has a setting for it.